src/core/qgsfields.cpp

```diff
@@ -403,15 +403,15 @@ QIcon QgsFields::iconForFieldType( QVariant::Type type, QVariant::Type subType,
 
 int QgsFields::lookupField( const QString &fieldName ) const
 {
-  if ( fieldName.isEmpty() ) //shortcut
-    return -1;
-
   for ( int idx = 0; idx < count(); ++idx )
   {
     if ( d->fields[idx].field.name() == fieldName )
       return idx;
   }
 
+  if ( fieldName.isEmpty() )
```

Comment:
This will prevent alias handling. Is it required?

Reply:
@m-kuhn , it's an optimization. We don't allow comparison against empty alias string (which makes sense since by default the alias string value is empty). So instead of going through needless for loops, we just bail out here.